osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/security/providers/HsmEncryptionProvider.java

```diff
@@ -45,63 +45,73 @@
 
     private final KeyStore keyStore;
 
-    public HsmEncryptionProvider(File keyStoreFile) {
+    public HsmEncryptionProvider(final File keyStoreFile) {
         try {
             super.setKeyFile(keyStoreFile);
             this.keyStore = KeyStore.getInstance(TYPE, PROVIDER);
-            FileInputStream fIn = new FileInputStream(keyStoreFile);
+            final FileInputStream fIn = new FileInputStream(keyStoreFile);
             this.keyStore.load(fIn, null);
-        } catch (CertificateException | NoSuchAlgorithmException | NoSuchProviderException | IOException | KeyStoreException e) {
+        } catch (final CertificateException | NoSuchAlgorithmException | NoSuchProviderException | IOException | KeyStoreException e) {
             throw new EncrypterException("Could not read keystore", e);
         }
     }
 
-    public Secret decrypt(EncryptedSecret secret, String keyReference) {
-
-        Secret decryptedSecret = super.decrypt(secret, keyReference);
-
-        byte[] decryptedSecretBytes = decryptedSecret.getSecret();
-
-        if (decryptedSecretBytes.length > KEY_LENGTH) {
-
-            byte[] truncatedDecryptedSecretBytes = Arrays.copyOfRange(decryptedSecretBytes, 0,
-                    decryptedSecretBytes.length-16);
-
-            LOGGER.trace("Truncating decrypted key from " + Hex.encodeHexString(decryptedSecretBytes) + " to " +
+    @Override
+    public byte[] decrypt(final EncryptedSecret secret, final String keyReference) {
+        byte[] decryptedSecret = super.decrypt(secret, keyReference);
+        if (decryptedSecret.length > KEY_LENGTH) {
+            final byte[] truncatedDecryptedSecretBytes = Arrays.copyOfRange(decryptedSecret, 0,
+                    decryptedSecret.length-16);
```

Comment:
What does this do, and why does it do this?
The code looks sketchy and raises questions. If this does what it is supposed to do, it might be good to add a few explanatory comments.

If the `decryptedSecret` has more than 16 bytes, it drops the last 16 bytes (if this should match `KEY_LENGTH` it would be better to use the constant in both cases, and not have the constant in one place and a literal 16 in the other).

What should be the result?
Something that has exactly `KEY_LENGHT` bytes, at most `KEY_LENGTH` bytes, or whatever comes in stripping the last 16 bytes.

This reminds me a bit of other code where the first number of bytes are left out if they are all zeros, but looks somewhat different.